nibabel/arrayproxy.py

```diff
@@ -125,6 +131,27 @@ def __init__(self, file_like, spec, mmap=True):
         # Permit any specifier that can be interpreted as a numpy dtype
         self._dtype = np.dtype(self._dtype)
         self._mmap = mmap
+        self._keep_file_open = keep_file_open
+        self._lock = Lock()
+
+    def __del__(self):
+        '''If this ``ArrayProxy`` was created with ``keep_file_open=True``,
+        the open file object is closed if necessary.
+        '''
+        if hasattr(self, '_opener') and not self._opener.closed:
```

Comment:
What happens if the user passes an open file to the original image creation routine?  Does that get closed too?